it/src/main/java/com/google/cloud/teleport/it/pubsub/PubsubUtils.java

```diff
@@ -0,0 +1,109 @@
+  private PubsubUtils() {}
+
+  /**
+   * Regular expression to validate/extract a topic name string, following
+   * projects/{project_id}/subscription/{topic_name}.
+   */
+  private static final Pattern TOPICS_PATTERN =
+      Pattern.compile("^projects/(?P<project_id>[^/]+)/topics/(?P<topic_name>[^/]+)$");
+
+  /**
+   * Regular expression to validate/extract a subscription name string, following
+   * projects/{project_id}/subscription/{subscription_name}.
+   */
+  private static final Pattern SUBSCRIPTIONS_PATTERN =
+      Pattern.compile(
+          "^projects/(?P<project_id>[^/]+)/subscriptions/(?P<subscription_name>[^/]+)$");
+
+  /**
+   * Convert the instance of {@link Topic} to a {@link TopicName}.
+   *
+   * <p>The idea of this project is to abstract the conversion/parse of the format
+   * <strong>projects/{project_id}/topics/{topic_name}</strong>
+   *
+   * @param topic Base topic.
+   * @return The reference to the topic.
+   */
+  public static TopicName toTopicName(Topic topic) {
+    Matcher matcher = TOPICS_PATTERN.matcher(topic.getName());
+    checkArgument(
+        matcher.find(),
+        "Topic name must be in the format 'projects/{project_id}/topics/{topic_name}.");
+
+    return TopicName.of(matcher.group("project_id"), matcher.group("topic_name"));
+  }
+
+  /**
+   * Convert the instance of {@link Subscription} to a {@link SubscriptionName}.
+   *
+   * <p>The idea of this project is to abstract the conversion/parse of the format
+   * <strong>projects/{project_id}/subscription/{subscription_name}</strong>
+   *
+   * @param subscription Base subscription.
+   * @return The reference to the subscription.
+   */
+  public static SubscriptionName toSubscriptionName(Subscription subscription) {
+    Matcher matcher = SUBSCRIPTIONS_PATTERN.matcher(subscription.getName());
+    checkArgument(
+        matcher.find(),
+        "Subscription name must be in the format 'projects/{project_id}/subscriptions/{subscription_name}.");
+
+    return SubscriptionName.of(matcher.group("project_id"), matcher.group("subscription_name"));
+  }
+
+  /**
+   * Creates a topic name. Based on {@link
+   * com.google.cloud.teleport.it.dataflow.DataflowUtils#createJobName(String)}
+   *
+   * <p>If there are uppercase characters in {@code prefix}, then this will convert them into a dash
+   * followed by the lowercase equivalent of that letter.
+   *
+   * <p>The topic name will normally be unique, but this is not guaranteed if multiple topics with
+   * the same prefix are created in a short period of time.
+   *
+   * @param prefix a prefix for the topic
+   * @return the prefix plus some way of identifying it separate from other topics with the same
+   *     prefix
+   */
+  public static String createTestId(String prefix) {
+    String convertedPrefix =
+        CaseFormat.UPPER_CAMEL.converterTo(CaseFormat.LOWER_HYPHEN).convert(prefix);
+    String formattedTimestamp =
+        DateTimeFormatter.ofPattern("yyyyMMddHHmmss")
```

Comment:
Could we maybe get this down to microsecond precision? That'll decrease the chance of collision. (Second precision is good too, but I think Pub/Sub can handle greater precision, and that'll further reduce the chance of collision.)